tests/s3.bats

```diff
@@ -43,6 +43,20 @@ load helpers.sh
   [[ "$(SHA $DIR/blob)" == "$(SHA $TMP_BLOB)" ]]
 }
 
+# List
+
+@test "listing keys from test bucket" {
+  run GET "$CACHE/list?bucket=$BUCKET"
+  [[ "$status" -eq 0 ]]
+  [[ "$output" == "200" ]]
+  [[ "$(cat $TMP_BLOB | jq '.keys | length')" == "3" ]]
+
+  run GET "$CACHE/list?bucket=$BUCKET&prefix=folder"
+  [[ "$status" -eq 0 ]]
+  [[ "$output" == "200" ]]
+  [[ "$(cat $TMP_BLOB | jq '.keys | length')" == "2" ]]
+}
+
```

Comment:
Would be good to also test the 404 case in case the prefix is non-existent.

Reply:
Done, added failure cases to all tests